src/tally.rs

```diff
@@ -33,61 +58,73 @@ async fn days() -> Result<(Vec<(PathBuf, u32)>, Vec<u32>), AocError>
         }
     }
 
-
     Ok((have, dont_have))
 }
 
-async fn build_day(day: u32) -> Result<(), AocError>
+async fn build_days(cargo_folder: PathBuf, days: &[(PathBuf, u32)]) -> Result<Vec<u32>, AocError>
 {
-    let bin = format!("day_{:02}", day);
-    let res = tokio::process::Command::new("cargo")
-        .args(["build", "--release", "--bin", &bin])
-        .output()
-        .await?;
+    let sty = ProgressStyle::with_template(
```

Comment:
I think we can extract this and the creating of a progress bar to a function that returns the progressbar with the given style. This way we can use that function here and in `run_days`. I think something like:

```rs
fn get_progressbar(len: u64) -> ProgressBar
{
    let sty = ProgressStyle::with_template(
        "[{elapsed_precise}] {msg}... {bar:40.cyan/blue} {pos:>7}/{len:7}",
    )
    .unwrap()
    .progress_chars("##-");

    ProgressBar::new(len).with_style(sty)
}
```

Feel free to change the function name or something, but I think a separate function could be nice though

Reply:
Good idea